File: sdks/java/src/main/java/com/puxbay/resources/CategoriesResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Category;

public class CategoriesResource {
    private final Puxbay client;

    public CategoriesResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Category> list(int page) throws PuxbayException {
        return (PaginatedResponse<Category>) client.request("GET", "categories/?page=" + page, null, PaginatedResponse.class);
    }

    public Category get(String categoryId) throws PuxbayException {
        return client.request("GET", "categories/" + categoryId + "/", null, Category.class);
    }

    public Category create(Category category) throws PuxbayException {
        return client.request("POST", "categories/", category, Category.class);
    }

    public Category update(String categoryId, Category category) throws PuxbayException {
        return client.request("PATCH", "categories/" + categoryId + "/", category, Category.class);
    }

    public void delete(String categoryId) throws PuxbayException {
        client.request("DELETE", "categories/" + categoryId + "/", null, Void.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/CategoriesResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Category
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class CategoriesResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1): PaginatedResponse<Category> =
        client.request(HttpMethod.Get, "categories/?page=$page")
    
    suspend fun get(categoryId: String): Category =
        client.request(HttpMethod.Get, "categories/$categoryId/")
    
    suspend fun create(category: Category): Category =
        client.request(HttpMethod.Post, "categories/", category)
    
    suspend fun update(categoryId: String, category: Category): Category =
        client.request(HttpMethod.Patch, "categories/$categoryId/", category)
    
    suspend fun delete(categoryId: String) {
        client.request<Unit>(HttpMethod.Delete, "categories/$categoryId/")
    }
}
